src/sentry/static/sentry/app/views/settings/project/projectServiceHookDetails.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import createReactClass from 'create-react-class';
+import {Box} from 'grid-emotion';
+
+import {t} from '../../../locale';
+import {
+  addErrorMessage,
+  addSuccessMessage,
+} from '../../../actionCreators/settingsIndicator';
+import AsyncView from '../../asyncView';
+import ApiMixin from '../../../mixins/apiMixin';
+import Form from '../components/forms/form';
+import JsonForm from '../components/forms/jsonForm';
+import serviceHookSettingsFields from '../../../data/forms/serviceHookSettings';
+import SettingsPageHeader from '../components/settingsPageHeader';
+
+const TOAST_DURATION = 10000;
+
+const ServiceHookSettingsForm = createReactClass({
+  displayName: 'ServiceHookSettingsForm',
+
+  propTypes: {
+    location: PropTypes.object,
+    orgId: PropTypes.string.isRequired,
+    projectId: PropTypes.string.isRequired,
+    hookId: PropTypes.string.isRequired,
+    access: PropTypes.object.isRequired,
+    initialData: PropTypes.object.isRequired,
+    onSave: PropTypes.func.isRequired,
+  },
+
+  mixins: [ApiMixin],
+
+  render() {
+    let {initialData, orgId, projectId, hookId, onSave, access} = this.props;
+
+    return (
+      <Form
+        apiMethod="PUT"
+        apiEndpoint={`/projects/${orgId}/${projectId}/hooks/${hookId}/`}
+        saveOnBlur
+        allowUndo
+        initialData={initialData}
+        onSubmitSuccess={(change, model, id) => {
```

Comment:
this was copy pasted from org settings

Reply:
There's a PR that refactors and provides this by default